enterprise/backend/src/metabase_enterprise/sso/integrations/sso_utils.clj

```diff
@@ -59,10 +60,12 @@
   reuse it."
   [user :- UserAttributes]
   (try
-    (u/prog1 (first (t2/insert-returning-instances! User (merge user {:password (str (random-uuid))})))
+    (u/prog1 (t2/insert-returning-instance! User (merge user {:password (str (random-uuid))}))
       (log/infof "New SSO user created: %s (%s)" (:common_name <>) (:email <>))
       ;; publish user-invited event for audit logging
-      (events/publish-event! :event/user-invited {:object (assoc <> :sso_source (:sso_source user))})
+      ;; skip sending user invited emails for sso users
+      (binding [events.notification/*skip-sending-notification?* true]
```

Comment:
Using a dynamic variable feels dangerous, we might refactor the notifications to run async on different threads.

What do you think about rather putting an option instead the event payload?

Reply:
the async change executes in a bound-fn, so it should preserve the dynamic var value. but yeah, I'll write a test.

Reply:
wrote a test: metabase-enterprise.sso.integrations.sso-utils-test/create-new-sso-user-do-not-send-email-test